vllm/v1/worker/gpu_model_runner.py

```diff
@@ -604,6 +607,54 @@ def profile_run(self) -> None:
             torch.tensor([], dtype=torch.float32, device=self.device)
             for _ in range(self.num_attn_layers)
         ]
+
+        # Profile with multimodal encoder & encoder cache.
+        # TODO (ywang96): generalize this beyond image modality since
+        # mm_input_mapper only supports image inputs.
+        if self.is_multimodal_model:
+
+            # Create dummy batch of multimodal inputs.
+            dummy_mm_data = self.input_registry \
+                .dummy_data_for_profiling(self.model_config,
+                                            self.max_num_tokens,
+                                            self.mm_registry).multi_modal_data
+            dummy_mm_kwargs, _ = self.mm_input_mapper.process_inputs(
+                mm_data=dummy_mm_data,
+                mm_hashes=None,
+                mm_processor_kwargs=None,
+                precomputed_mm_inputs=None)
+
+            # NOTE: Currently model is profiled with a single non-text
+            # modality even when it supports multiple.
+            max_tokens_per_mm_item = max(
+                self.mm_registry.get_max_tokens_per_item_by_modality(
+                    self.model_config).values())
+
+            max_num_mm_items = min(
+                self.max_num_encoder_input_tokens,
+                self.encoder_cache_size) // max_tokens_per_mm_item
```

Comment:
A dumb question: in our current implementation, shouldn't `encoder_cache_size` be always larger than or equal to `max_num_encoder_input_tokens`?

Reply:
Why should the encoder cache size affect the maximum tokens we can process as input? It seems we should make the decision if we can allow the cache size to be smaller than the max number of encoder input tokens

Reply:
@ywang96 @mgoin IIUC, we always stores the encoder outputs to the encoder cache, and then use them in the decoder. Therefore, if the encoder cache is smaller than that of the encoder outputs (which is in turn equal to the number of the encoder inputs).

Reply:
Since both values are hardcoded to 16384 anyways, I will merge this PR as is and revisit this particular logic later when we tune the space & compute budget